### ISSUE TITLE
[SWIG] <SWIG>/<4.0.2>: pcre is disabled for SWIG 4.0.1, 4.0.2, 4.1.0, 4.1.1

### ISSUE THREAD

**Issue by jonananas** [conan-io/conan-center-index#25106]
### Description

As of the latest push to conancenter 2024-07-01 pcre is disabled for all versions. This was not the case with the previous recipe, and I believe pcre should be enabled since it is listed as a requirement.
- pcre is listed as a requirement for 4.0.1 and 4.0.2
- pcre2 is listed as a requirement for 4.1.0 and 4.1.1
- pcre is then disabled in https://github.com/conan-io/conan-center-index/blob/26068610609a30566ebd724e1a7c5513215d1bce/recipes/swig/all/conanfile.py#L144

This affects all platforms, the effect can be seen clearly by doing `swig -version`, -pcre should be +pcre:
```bash
> swig -version
SWIG Version 4.1.1

Compiled with clang++ [x86_64-apple-darwin22.1.0]

Configured options: -pcre

Please see https://www.swig.org for reporting bugs and further information
```

Note: We are still on conan 1, but the steps are reproducable using conan 2.

### Package and Environment Details

* Package Name/Version: **swig/4**
* Operating System+version: **

### Conan profile

df

### Steps to reproduce

cd recipes/swig/all
conan create . swig/4.0.2@algoryx/stable-brick --build=missing
# will run swig -version when done, outputting -pcre which should be +pcre
# same for all supported versions in recipe

(Alternatively install swig from conan and run `swig -version`)

### Logs

df

**Comment by jcar87** [comment]
Hi @jonananas  - thanks for reporting this.

I have had a look at the recipe and it is confusing -
this line ` replace_in_file(self, os.path.join(self.source_folder, "configure.ac")` disables looking for pcre2 with pkg-config, but relies on Conan still propagating the flags.
 So the build system is recording that it was built "without" pcre, but it may actually be build with it - however, it's unclear to me.
 
Have you encountered a situation where pcre2 support actually does not work and results in an error? It will be easier to troubleshoot from there.

**Comment by valgur** [comment]
> Hi @jonananas - thanks for reporting this.
> 
> I have had a look at the recipe and it is confusing - this line ` replace_in_file(self, os.path.join(self.source_folder, "configure.ac")` disables looking for pcre2 with pkg-config, but relies on Conan still propagating the flags. So the build system is recording that it was built "without" pcre, but it may actually be build with it - however, it's unclear to me.
> 
> Have you encountered a situation where pcre2 support actually does not work and results in an error? It will be easier to troubleshoot from there.

Not pkg-config. It was using a custom `pcre2-config` executable provided by `pcre2` for the package info and build flags. I had to swap it out at the time because it was reporting incorrect library names for static libs. This has been fixed since then (#20586) and the workaround in `swig` can probably be reverted.

**Comment by jonananas** [comment]
> Hi @jonananas - thanks for reporting this.
> 
> I have had a look at the recipe and it is confusing - this line ` replace_in_file(self, os.path.join(self.source_folder, "configure.ac")` disables looking for pcre2 with pkg-config, but relies on Conan still propagating the flags. So the build system is recording that it was built "without" pcre, but it may actually be build with it - however, it's unclear to me.
> 
> Have you encountered a situation where pcre2 support actually does not work and results in an error? It will be easier to troubleshoot from there.

Yes, we got failing builds after clearing our conan cache with the errors:

SWIG(999999): error : PCRE regex matching is not available in this SWIG build.
SWIG(33): error : PCRE regex matching is not available in this SWIG build.

Indicating that SWIG /* !HAVE_PCRE */, from SWIG source:

https://github.com/swig/swig/blob/master/Source/Swig/naming.c#L1283C7-L1283C23

**Comment by jonananas** [comment]
> Not pkg-config. It was using a custom `pcre2-config` executable provided by `pcre2` for the package info and build flags. I had to swap it out at the time because it was reporting incorrect library names for static libs. This has been fixed since then (#20586) and the workaround in `swig` can probably be reverted.

Are there any plans for fixing this? I'll be happy to push a PR for dropping the workaround, but while it solves our problem, I'm not certain it doesn't have any other unwanted impacts?
